common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -125,8 +127,16 @@ class @Problem
         @forceUpdate response
         delete window.queuePollerID
       else
-        # TODO: Some logic to dynamically adjust polling rate based on queuelen
-        window.queuePollerID = window.setTimeout(@poll, 1000)
+        new_timeout = prev_timeout * 2
+        # if the timeout is greather than 1 minute
+        if new_timeout >= 60000
+          delete window.queuePollerID
+          @gentle_alert "The grader is taking too long to respond. Please refresh this page if you would like updates."
```

Comment:
@dianakhuang How about: "The grading process is still running. Refresh the page to see updates."